modules/runner/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItDistributionZonesTest.java

```diff
@@ -0,0 +1,258 @@
+        } catch (Exception e) {
+            assertInstanceOf(ExecutionException.class, e);
+
+            assertInstanceOf(ReplicaUnavailableException.class, e.getCause());
+        }
+
+        cluster.knockOutNode(2, PARTITION_NETWORK);
+
+        assertTrue(waitAssingments(List.of(
+                Set.of(0, 1, 3),
+                Set.of(0, 1, 3),
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 3)
+        )));
+
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(0).node()).get());
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(1).node()).get());
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(3).node()).get());
+
+        cluster.reanimateNode(2, PARTITION_NETWORK);
+
+        assertTrue(waitAssingments(List.of(
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2)
+        )));
+
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(0).node()).get());
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(1).node()).get());
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(2).node()).get());
+
+        try {
+            table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(3).node()).get();
+
+            fail();
+        } catch (Exception e) {
+            assertInstanceOf(ExecutionException.class, e);
+
+            assertInstanceOf(ReplicaUnavailableException.class, e.getCause());
+        }
+    }
+
+    private boolean waitAssingments(List<Set<Integer>> nodes) throws InterruptedException {
+        return waitForCondition(() -> {
+            for (int i = 0; i < nodes.size(); i++) {
+                Set<Integer> excpectedAssignments = nodes.get(i);
+
+                ExtendedTableConfiguration table =
+                        (ExtendedTableConfiguration) cluster.node(i)
+                                .clusterConfiguration().getConfiguration(TablesConfiguration.KEY).tables().get("TEST");
+
+                byte[] assignmentsBytes = table.assignments().value();
+
+                Set<String> assignments;
+
+                if (assignmentsBytes != null) {
+                    assignments = ((List<Set<Assignment>>) ByteUtils.fromBytes(assignmentsBytes)).get(0)
+                            .stream().map(assignment -> assignment.consistentId()).collect(Collectors.toSet());
+                } else {
+                    assignments = Collections.emptySet();
+                }
+
+                LOG.info("Assignments for node " + i + ": " + assignments);
+
+                if (!(excpectedAssignments.size() == assignments.size())
+                        || !excpectedAssignments.stream().allMatch(node -> assignments.contains(cluster.node(node).name()))) {
```

Comment:
It's not a good idea to check whether node name contains the number and consider that as an equality relation. I'd use full node names for an expected value.